addons/udes_ldap/models/res_company_ldap.py

```diff
@@ -0,0 +1,52 @@
+"""
+UDES LDAP configuration model.
+
+In this model we override the standard authentication approach by using the
+credentials of the logging-in user to search for them selves on the LDAP
+server.  This removes the need for a service account to perform the search.
+"""
+
+from ldap.dn import escape_dn_chars
+from odoo import models, fields
+
+
+class CompanyLDAP(models.Model):
+    _inherit = "res.company.ldap"
+
+    u_ldap_binddn_fmt = fields.Char(
+        string="LDAP binddn format",
+        help="A format string for the Distinguished Name used to bind to the"
+        "LDAP server.  The user's uid will be formatted into this string.",
+        default="",
+        required=True,
+    )
+
+    def get_ldap_dicts(self):
+        """Add our custom field to the conf dict and removed unused fields."""
+        res = super().get_ldap_dicts()
+        for rec in res:
+            ldap = self.sudo().browse(rec["id"])
+            rec["u_ldap_binddn_fmt"] = ldap.u_ldap_binddn_fmt
+            del rec["ldap_binddn"]
```

Comment:
Is this "list" ever going to change? Do we want to use a list + loop instead of `del` twice?

Reply:
There could be more than one LDAP configured, in theory.  Or are you suggesting an inner loop to drop the keys that we don't want?

Another option might be to mark those fields as [deprecated](https://github.com/unipartdigital/odoo/blob/ee08a6f63711c950bc6c3eae61fd7f959824c0c4/odoo/fields.py#L310), which would [prevent them being retrieved](https://github.com/unipartdigital/odoo/blob/ee08a6f63711c950bc6c3eae61fd7f959824c0c4/odoo/fields.py#L446), but that doesn't seem to be very common :grimacing:

Reply:
Yes, I was suggesting inner loop to drop the keys we don't want.

Reply:
Done